### ISSUE TITLE
channel 2: open failed: unknown channel type: unsupported channel type

### ISSUE THREAD

**Issue by stefanfritsch** [okteto/remote-kubernetes#146]
Hi,

I get the error from the subject when trying to use the VSCode extension to a private cluster.

**Describe the bug**

The connection to the remote fails with the following error:

```
[15:19:05.598] > Warning: Permanently added '[localhost]:50192' (RSA) to the list of known hosts.
> SSH_AUTH_SOCK=/tmp/auth-agent926329077/listener.sock
> Connected to SSH Host - Please do not close this terminal
> channel 2: open failed: unknown channel type: unsupported channel type
> channel 2: open failed: unknown channel type: unsupported channel type
> channel 2: open failed: unknown channel type: unsupported channel type
```

**To Reproduce**

1. Install okteto in vscode
2. clone the [Okteto hello-world](https://github.com/okteto/vscode-remote-go) repo
3. Run Okteto Up
4. I also tried with a reduced okteto.yml with less capabilities:

```
name: hello-world
image: okteto/golang:1.16
autocreate: true
workdir: /okteto
command: ["bash"]
```

**Result**

1. The deployment is launched
2. Okteto opens a new VSCode window that connects successfully until the error shown above occurs
3. The deployment works fine. `kubectl exec -ti hello-world -- bash` is also successful
4. In the local VSCode window there are no errors:

```
PS C:\Users\StefanFritsch\Documents\spielwiese\vscode-remote-go> C:\Users\StefanFritsch\AppData\Local\Programs\okteto.exe up -f 'c:\Users\StefanFritsch\Documents\vscode-remote-go\okteto.yml' --remote '22100' --loglevel=warn
 +  Persistent volume successfully attached
 +  Images successfully pulled
 +  Files synchronized
    Context:  default
    Namespace: hello-world
    Name:      hello-world   
```

**Expected behavior**

A successful connection

**Desktop (please complete the following information):**

- OS: Windows 10 
- Version 21H1

Thank you,
Stefan

**Comment by waddahAldrobi** [comment]
How did you get your Okteta:Up to work? Mine fails at that command.

![image](https://user-images.githubusercontent.com/28662355/129070884-bb18e651-3ddf-45cc-903a-33aa6b671c1f.png)

**Comment by stefanfritsch** [comment]
Easy: It just worked....until it didn't :)

**Comment by rberrelleza** [comment]
thanks for reporting, we're looking into it!  What  kubernetes version distribution are you using @stefanfritsch @waddahAldrobi ?

**Comment by stefanfritsch** [comment]
@rberrelleza thanks for looking into it.

The cluster was created with rke 1.2

**Comment by stefanfritsch** [comment]
I also tried it with a k3s cluster. Same issue

**Comment by rberrelleza** [comment]
we haven't been able to replicate this issue. Could you try and see if it still reproduces with the version 0.3.4 of the extension? If it does reproduce, could you share your $HOME/.ssh/config file with us, as well as the version of your SSH client?

**Comment by stefanfritsch** [comment]
Sorry this took so long:

### ssh-version

```
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     ssh.exe                                            8.1.0.1    C:\WINDOWS\System32\OpenSSH\ssh.exe
```

### .ssh/config

And the entry in .ssh/config is

```
# entry generated by okteto
Host hello-world.okteto
  ForwardAgent yes
  PubkeyAcceptedKeyTypes +ssh-rsa
  HostName localhost
  Port 22100
  StrictHostKeyChecking no
  UserKnownHostsFile /dev/null
  IdentityFile "C:\Users\StefanFritsch\.okteto\id_rsa_okteto"
```

Everything else in .ssh/config are other host directives that do not conflict with this one.

## New developments

### ssh-connection

The ssh-connection itself is fine. I can connect to the container from a powershell window using the `.ssh/config` entry. I 

```
PS C:\Users\StefanFritsch> ssh hello-world.okteto
Warning: Permanently added '[localhost]:22100' (RSA) to the list of known hosts.
Welcome to your development container. Happy coding!
stefan-fritsch:hello-world okteto>
```

### Error message

The error message has changed slightly since the last time. It now also complains about "channel 3", whatever that is.

```
Warning: Permanently added '[localhost]:22100' (RSA) to the list of known hosts.
SSH_AUTH_SOCK=/tmp/auth-agent230119601/listener.sock
Connected to SSH Host - Please do not close this terminal
channel 2: open failed: unknown channel type: unsupported channel type
channel 3: open failed: unknown channel type: unsupported channel type
channel 2: open failed: unknown channel type: unsupported channel type
```

The following error notification is also new (although I'm not yet 100% certain it's okteto as some extensions just got updated):

```
Failed to connect to the remote extension host server (Error: WebSocket close with status code 1006)
```

**Comment by rberrelleza** [comment]
This is fixed in the 0.4.1 release

**Comment by stefanfritsch** [comment]
@rberrelleza today I had the opportunity to try the extension again and it's still broken in 0.4.3. Both with Windows ssh 8.9 as well as with git ssh 8.8 (see below)

Same issue as before:

### The ssh tunnel terminal
```
Warning: Permanently added '[localhost]:22100' (RSA) to the list of known hosts.
Connected to SSH Host - Please do not close this terminal
channel 2: open failed: unknown channel type: unsupported channel type
channel 3: open failed: unknown channel type: unsupported channel type
channel 2: open failed: unknown channel type: unsupported channel type
```

### The javascript console in the developer tools
```
ERR WebSocket close with status code 1006: Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (vscode-file://vscode-app/c:/Users/StefanFritsch/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:611:39219)
```

### Current OS

* OS: Windows 11 21H2
* SSH: OpenSSH_for_Windows_8.9p1, LibreSSL 3.4.3

I also tried it using the ssh of git bash (C:\Program Files\Git\usr\bin\ssh.exe):

* SSH: OpenSSH_8.8p1

With WSL2 it doesn't work because okteto only listens on localhost and is not accessible from WSL2 afaics.
